src/utils/formatTime.ts

```diff
@@ -1,20 +1,31 @@
 import { BigNumber, BigNumberish } from '@ethersproject/bignumber'
 
-export function detailedTimeString(endTime?: BigNumberish) {
-  const secondsLeft =
-    BigNumber.from(endTime).sub(Math.floor(Date.now().valueOf())).toNumber() /
-    1000
-
-  const days = secondsLeft / 60 / 60 / 24
-  const hours = (secondsLeft / 60 / 60) % 24
-  const minutes = (secondsLeft / 60) % 60
-  const seconds = secondsLeft % 60
-
-  return (
-    `${days && days > 1 ? Math.floor(days).toString() + 'd ' : ''}${
-      hours && hours >= 1 ? Math.floor(hours) + 'h ' : ''
-    }
-      ${days < 1 && minutes >= 1 ? Math.floor(minutes) + 'm ' : ''}
-      ${minutes < 1 ? Math.floor(seconds) + 's' : ''}` || '--'
+import { SECONDS_IN_DAY } from 'constants/numbers'
+
+export function detailedTimeString(timeSeconds?: BigNumberish) {
+  const timeSecondsNumber = BigNumber.from(timeSeconds).toNumber()
+
+  const days = Math.floor(timeSecondsNumber / 60 / 60 / 24)
+  const hours = (timeSecondsNumber / 60 / 60) % 24
+  const minutes = (timeSecondsNumber / 60) % 60
+  const seconds = timeSecondsNumber % 60
+
+  const daysText = `${days && days > 0 ? days.toString() + 'd ' : ''}`
```

Comment:
There was a slight bug in this before - if you passed endTime = 86400, it would return `0s` when it should return `1d`.